stowsh

```diff
@@ -50,6 +50,36 @@ stowsh_setpaths() {
         deperr "GNU findutils"
         return 1
     fi
+    if [[ "${USEGIT}" -eq 1 ]]; then
+        flist="list_git_files"
+    else
+        flist="list_files"
+    fi
+}
+
+list_files() {
+    if [[ "$1" == "-d" ]];
+    then
+        $findcmd . -mindepth 1 -type d -printf "%P\0"
+    else
+        $findcmd . -type f -printf "%P\0" -or -type l -printf "%P\0"
+    fi
+}
+
+list_git_files() {
+    gitfiles=( `git ls-files` )
+    if [[ "${#gitfiles[@]}" == 0 ]]; then return; fi
+
+    if [[ "$1" == "-d" ]];
+    then
+        printf "%s\n" ${gitfiles[@]} |\
```

Comment:
I think you need double quotes around `${gitfiles[@]}`.

Reply:
I think it's not necessary: `xargs` already picks one folder per line. Anyway, just to be on the safe part, I added the double quotes and squashed.

Reply:
Thanks. My general rule is: if [shellcheck](https://github.com/koalaman/shellcheck/) tells me to do something, I do it ;-)